src/tests/test_types.cpp

```diff
@@ -264,4 +264,109 @@ INSTANTIATE_TEST_SUITE_P(TestStringSuite, TestString,
                                            TestStringParam("abc", ""),
                                            TestStringParam("bb", "aa")));
 
+using DatePtr = typename std::unique_ptr<std::byte[], mo::DateTypeDeleter>;
+
+DatePtr DatePtrFromString(std::string const& s, std::unique_ptr<mo::DateType> const& date_type) {
+    return {date_type->MakeValue(boost::gregorian::from_simple_string(s)),
+            date_type->GetDeleater()};
+}
+
+TEST(TestDateSuite, Compare) {
+    std::unique_ptr<mo::DateType> date_type(
+            mo::CreateSpecificType<mo::DateType>(mo::TypeId::kDate, true));
+    auto test = [date_type = std::move(date_type)](
+                        std::string const& date1, std::string const& date2, mo::CompareResult res) {
+        auto ptr1 = DatePtrFromString(date1, date_type);
+        auto ptr2 = DatePtrFromString(date2, date_type);
```

Comment:
Type is not obvious from the right side, so use it explicitly instead of relying on the type deduction